### PR TITLE
Add TAPAS MLM-only models

### LINKED ISSUE #12916: fill-mask pipeline with tables (TapasForMaskedLM) fails DataFrame type assertion
## Environment info

- `transformers` version: 4.4.0
- Platform: macOS
- Python version: 3.9.2
- PyTorch version (GPU?): 1.8.9 (N/A)
- Tensorflow version (GPU?): N/A
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help

@LysandreJik 
@NielsRogge

## Information

Model I am using: **TapasForMaskedLM**

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [ X ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1. Slight modification of [example](https://huggingface.co/transformers/v4.5.1/model_doc/tapas.html#tapasformaskedlm) to include `fill-mask` pipeline
2. Script to run:
```
if __name__ == '__main__':

    from transformers import TapasConfig,TapasTokenizer,TapasForMaskedLM
    from transformers import pipeline
    import pandas as pd
    import numpy as np
    import torch
    import sys

    config = TapasConfig.from_pretrained(
        'google/tapas-base-finetuned-wtq',from_pt=True)
    model = TapasForMaskedLM.from_pretrained(
        'google/tapas-base-finetuned-wtq', config=config)
    tokenizer=TapasTokenizer.from_pretrained(
        "google/tapas-base-finetuned-wtq", from_pt=True)

    data= {
        "actors": ["brad pitt", "leonardo di caprio", "george clooney"],
        "age": ["56", "45", "59"],
        "number of movies": ["87", "53", "69"],
        "date of birth": ["7 february 1967", "10 june 1996", "28 november 1967"]
    }

    table = pd.DataFrame.from_dict(data)

    queries=[
        f"The number of movies Brad Pitt acted in is {tokenizer.mask_token}",
        f"Leonardo di caprio's age is {tokenizer.mask_token}"]

    nlp = pipeline(task="fill-mask",framework="pt",model=model, tokenizer=tokenizer)
    test = nlp(queries, table=table)
```

3. From a short debugging it seems that `pandas/core/frame.py` is called and the following code overwrites `table` to a list:

```
        if isinstance(data, DataFrame):
            data = data._mgr

        if isinstance(data, BlockManager):
            if index is None and columns is None and dtype is None and copy is False:
                # GH#33357 fastpath
                NDFrame.__init__(self, data)
                return
```

## Expected behavior

Input table should not be overwritten to a list. Is this call to `frame.py` expected? If not what is the required steps to overcome this?

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

As requested by #12916, I've converted the TAPAS checkpoints which were pre-trained on masked language modeling (MLM) only. `TapasForMaskedLM` was previously defined, but there were no actual checkpoints available with the language modeling head. They're available now on the hub:

* [google/tapas-large-masklm](https://huggingface.co/google/tapas-large-masklm)
* [google/tapas-base-masklm](https://huggingface.co/google/tapas-base-masklm)
* [google/tapas-medium-masklm](https://huggingface.co/google/tapas-medium-masklm)
* [google/tapas-small-masklm](https://huggingface.co/google/tapas-small-masklm)
* [google/tapas-mini-masklm](https://huggingface.co/google/tapas-mini-masklm)
* [google/tapas-tiny-masklm](https://huggingface.co/google/tapas-tiny-masklm).

It also cleans up the conversion script of TAPAS a bit.

Fixes #12916

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Super cool, thanks for adding @NielsRogge!

**Comment by pafitis** [comment]
Thanks for this @NielsRogge !
